proposed/2021/NuGetConsumePDBInPackageReference.md

```diff
@@ -0,0 +1,216 @@
+In some cases, consuming .pbd, .xml files are needed to support certain features, like source link ( see issue [1458](https://github.com/dotnet/sdk/issues/1458)).
+The established design for consuming files out of packages is that NuGet tells the .NET SDK which files are important through the assets file, or to be specific, the [LockFileTargetLibrary](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileTargetLibrary.cs). 
+Currently, NuGet does not select pdb or xml files.
+
+In comparison, in packages.config, .pdb, .xml files are automatically copied from lib and runtime folder into the build output folder because of the [ResolveAssemblyReference](https://github.com/dotnet/msbuild/blob/main/src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs#L75) task. So there is no such problem in packages.config.
+
+## Who are the customers
+
+Partner team(.NET SDK) who needs to consume .pdb, .xml files for their features in PackageReference world. 
+
+Take source link as an example. When a NuGet package includes a .pdb file in its lib folder, in order to debug into the NuGet package, it needs the .pdb file to be copied to the build output folder of the project which reference the NuGet package.
+
+## Requirements
+
+* A feature to allow the .NET SDK to consume .pdb, .xml files.
+
+## Goals
+
+* Define an approach to make .NET SDK be able to consume .pdb, .xml files.
+
+## Non-Goals
+
+* .NET SDK side change to consume .pdb or .xml.
+
+## Solution
+
+For any given assembly under lib and runtime folder from a package reference, if there are files next to it that differ only by extension, NuGet will add a "related" property underneath the assembly in targets section of assets file, listing the extensions of these files.
+
+* Applies to all the compile time assemblies and runtime assemblies.
+* Apply to transitive dependencies.
+* Only apply to package reference. Will not apply to project reference.
+
+
+### Solution - Technical details
+
+#### Package reference:
+For example, one project has a package reference of PackageA. Suppose in folder ./PackageA/1.0.0/lib/netstandard2.0, there are PackageA.pdb, PackageA.xml file besides PackageA.dll.
+Currently, the targets section in assets file is:
+
+```json
+{
+  "version": 3,
+  "targets": {
+    ".NETCoreApp,Version=v5.0": {
+      "PackageA/1.0.0": {
+        "type": "package",
+        "compile": {
+           "lib/netstandard2.0/PackageA.dll": {}
+        },
+        "runtime": {
+           "lib/netstandard2.0/PackageA.dll": {}
+      }
+    }
+  },
+```
+When this solution is applied, the targets section in assets file will be:
+
+```json
+{
+  "version": 3,
+  "targets": {
+    ".NETCoreApp,Version=v5.0": {
+      "PackageA/1.0.0": {
+        "type": "package",
+        "compile": {
+           "lib/netstandard2.0/PackageA.dll": {
+              "related": "pdb,xml"
```

Comment:
Should we use a semicolon (`;`) as a separator here instead of a comma (`,`)?

Reply:
That's probably a better idea. 

`Aliases` uses `,`, but that's largely just following the precedent set by ProjectReference.Aliases.

Reply:
Thanks @dsplaisted and @nkolev92! Updated to use `;` as the delimiter.